Hudl.Mjolnir/Command/Command.cs

```diff
@@ -313,18 +313,30 @@ public TResult Invoke()
             }
             catch (Exception e)
             {
+                var tokenSourceCancelled = cancellationTokenSource.IsCancellationRequested;
                 executeStopwatch.Stop();
                 status = StatusFromException(e);
-
-                var instigator = new CommandFailedException(e, status).WithData(new
-                {
-                    Command = Name,
-                    Timeout = Timeout.TotalMilliseconds,
-                    Status = status,
-                    Breaker = BreakerKey,
-                    Pool = PoolKey,
-                });
-
+                // If the timeout cancellationTokenSource was cancelled and we got an TaskCancelledException here then this means the call actually timed out.
+                // Otherwise an TaskCancelledException would have been raised if a user CancellationToken was passed through to the method call, and was explicitly
+                // cancelled from the client side.
+                var instigator = status == CommandCompletionStatus.Canceled
+                    ? new CommandFailedException(e, status).WithData(new
+                    {
+                        Command = Name,
+                        Timeout = Timeout.TotalMilliseconds,
+                        Status = status,
+                        CommandCancellationReason = tokenSourceCancelled ? CommandCancellationReason.TimeoutCancellation : CommandCancellationReason.CallerTokenCancellation,
```

Comment:
If we wanted to DRY this up a bit, we could just make this field a "FailureReason" or something (i.e. not timeout-specific) and set it to "Faulted" for non-cancellations.